cluster-autoscaler/cloudprovider/builder/cloud_provider_builder.go

```diff
@@ -48,8 +49,15 @@ var AvailableCloudProviders = []string{
 const DefaultCloudProvider = gce.ProviderNameGCE
 
 // NewCloudProvider builds a cloud provider from provided parameters.
-func NewCloudProvider(opts config.AutoscalingOptions, do cloudprovider.NodeGroupDiscoveryOptions, rl *cloudprovider.ResourceLimiter) cloudprovider.CloudProvider {
+func NewCloudProvider(opts config.AutoscalingOptions) cloudprovider.CloudProvider {
```

Comment:
I wonder if it wouldn't be preferable to keep explicitly passing ResourceLimiter? I know we construct it from options anyway, but it doesn't feel like building it should be the responsibility of NewCloudProvider. It would make it easier if we ever need to customize RL and also marginally cleaner to mock. WDYT?

Reply:
Not feeling particularly strongly about it, but to me initializing ResourceLimiter here seems cleaner, since it's owned by CloudProvider object anyway. As for customizing and/or mocking, it isn't currently hidden behind any interface, so there's no way to substitute a custom struct/mock in a transparent way anyway.

Reply:
Fair point about mocking, I guess the difference in our opinions is because I'm still not thinking about RL as a part of CP, more like something separate we put in CP as a hack. Since we're not going to refactor RL again I probably should just get used to it.
I'm fine with leaving it as is.